hack/tools/logcheck/main.go

```diff
@@ -110,3 +119,40 @@ func isUnstructured(fName string) bool {
 
 	return false
 }
+
+// isKeysValid check if all keys in keyAndValues is string type
+func isKeysValid(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string) {
+	if len(keyValues)%2 != 0 {
+		pass.Report(analysis.Diagnostic{
+			Pos:     fun.Pos(),
+			Message: fmt.Sprintf("Invalid Number of arguments for %s", funName),
```

Comment:
I agree, if number of arguments doesn't match, we cannot be sure about what is missing. It could be value at the end of arguments, but also key in the middle of them. 
This makes analysis that assumes that keys are positioned in certain way hard to understand. In example provided by @adisky we would say tath `pod` should be a string, which doesn't make sense.

Reply:
get, will fixed on next push, Thanks for advice~